pkg/utils/node/predicates.go

```diff
@@ -18,17 +18,23 @@ import (
 	"time"
 
 	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha3"
-	"github.com/awslabs/karpenter/pkg/utils/pod"
 	v1 "k8s.io/api/core/v1"
 )
 
-func IsReadyAndSchedulable(node v1.Node) bool {
-	for _, condition := range node.Status.Conditions {
-		if condition.Type == v1.NodeReady {
-			return condition.Status == v1.ConditionTrue && !node.Spec.Unschedulable
-		}
+func IsReadyAndSchedulable(node *v1.Node) bool {
+	return getNodeCondition(node.Status.Conditions, v1.NodeReady).Status == v1.ConditionTrue && !node.Spec.Unschedulable
+}
+
+func IsReady(node *v1.Node) bool {
+	return getNodeCondition(node.Status.Conditions, v1.NodeReady).Status == v1.ConditionTrue
+}
+
+func FailedToJoin(node *v1.Node, duration time.Duration) bool {
```

Comment:
I might call this "grace period"

Now that I think about it, what do you think about using the ObjectMeta graceperiodseconds field?
https://github.com/kubernetes/apimachinery/blob/v0.21.3/pkg/apis/meta/v1/meta.go#L159